cardano-cli/src/Cardano/CLI/Compatible/Exception.hs

```diff
@@ -41,31 +42,37 @@ instance Exception CustomCliException where
 -- | Wrapper function which allows throwing of types of instance `Error`, attaching call stack
 -- from the call site
 throwCliError
-  :: forall e m a
-   . (HasCallStack, Show e, Typeable e, Error e, MonadIO m)
+  :: (HasCallStack, Show e, Typeable e, Error e, MonadIO m)
   => e
   -> m a
 throwCliError = withFrozenCallStack $ throwIO . CustomCliException
 
 fromEitherCli
-  :: forall e m a
-   . (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
+  :: (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
   => Either e a
   -> m a
 fromEitherCli = withFrozenCallStack $ \case
   Left e -> throwCliError e
   Right a -> return a
 
 fromEitherIOCli
-  :: forall e m a
-   . (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
+  :: (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
   => IO (Either e a)
   -> m a
 fromEitherIOCli action = withFrozenCallStack $ liftIO action >>= fromEitherCli
 
 fromExceptTCli
-  :: forall e m a
-   . (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
+  :: (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
   => ExceptT e IO a
   -> m a
 fromExceptTCli = withFrozenCallStack $ fromEitherIOCli . runExceptT
+
+handleFileErrorCli
```

Comment:
Why do we need this function? You can use `fromEitherIOCli @(FileError ())` instead. This seems to be a needless specialisation.